src/chpasswd.c

```diff
@@ -557,7 +557,7 @@ int main (int argc, char **argv)
 		} else
 #endif				/* USE_PAM */
 		{
-		const struct spwd *sp;
+		const struct spwd *sp = NULL;
```

Comment:
This is wrong.  I'm checking if the report is valid, which I'm not sure, but this initialization doesn't seem to do anything good, AFAICS.

Reply:
I confirm: the static analyzer seems to be wrong.  From the information I can see in the report, there's no use of uninitialized data (or I'm missing it).  It looks like a false positive.

Reply:
The report is valid, but I misunderstood it. The variable contains some members that are uninitialized, and that's what it's reporting. I must say that the report is not very accurate.

Reply:
Yeah, that's what I suspected, but since the members are not being used, it's not something we should care.  (At least, I didn't see any use of a member being uninitialized.)